terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts

```diff
@@ -46,69 +62,702 @@ beforeEach(() => {
+          .toDate(),
+        ghRunnerId: '33',
+      };
+      const matchRunner: GhRunners = [
+        {
+          id: 1,
+          name: ec2runner.instanceId,
+          os: 'linux',
+          status: 'busy',
+          busy: true,
+          labels: [],
+        },
+      ];
+      const mockedListRunners = mocked(listRunners).mockResolvedValue([]);
+      mocked(listGithubRunnersRepo).mockResolvedValue(matchRunner);
+
+      await scaleDown();
+
+      expect(mockedListRunners).toBeCalledWith(metrics, { environment: environment });
+    });
+
+    it('ec2runner with repo = undefined && org = undefined', async () => {
+      mocked(listRunners).mockResolvedValue([
+        {
+          instanceId: 'WG113',
+          launchTime: moment(new Date())
+            .subtract(minimumRunningTimeInMinutes + 5, 'minutes')
+            .toDate(),
+        },
+      ]);
+      const mockedResetRunnersCaches = mocked(resetRunnersCaches);
+      const mockedResetGHRunnersCaches = mocked(resetGHRunnersCaches);
+      const mockedResetSecretCache = mocked(resetSecretCache);
+      const mockedListGithubRunners = mocked(listGithubRunnersRepo);
+      const mockedListGithubRunnersOrg = mocked(listGithubRunnersOrg);
+      const mockedRemoveGithubRunnerOrg = mocked(removeGithubRunnerOrg);
+      const mockedRemoveGithubRunnerRepo = mocked(removeGithubRunnerRepo);
+      const mockedTerminateRunner = mocked(terminateRunner);
+
+      await scaleDown();
+
+      expect(mockedResetRunnersCaches).toBeCalledTimes(1);
+      expect(mockedResetGHRunnersCaches).toBeCalledTimes(1);
+      expect(mockedResetSecretCache).toBeCalledTimes(1);
+      expect(mockedListGithubRunners).not.toBeCalled();
+      expect(mockedListGithubRunnersOrg).not.toBeCalled();
+      expect(mockedRemoveGithubRunnerOrg).not.toBeCalled();
+      expect(mockedRemoveGithubRunnerRepo).not.toBeCalled();
+      expect(mockedTerminateRunner).not.toBeCalled();
+    });
+  });
+
+  describe('org', () => {
+    const environment = 'environment';
+    const scaleConfigRepo = 'test-infra';
+    const theOrg = ' a-owner';
+    const dateRef = moment(new Date());
+    const runnerTypes = new Map([
+      ['ignore-no-org-no-repo', { is_ephemeral: false } as RunnerType],
+      ['ignore-no-org', { is_ephemeral: false } as RunnerType],
+      ['keep-all-4', { is_ephemeral: false } as RunnerType],
+      ['a-ephemeral-runner', { is_ephemeral: true } as RunnerType],
+      ['keep-min-runners-oldest', { is_ephemeral: false } as RunnerType],
+      ['keep-lt-min-no-ghrunner', { is_ephemeral: false } as RunnerType],
+    ]);
+    const ghRunners = [
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+      { id: '0001', name: 'keep-this-not-min-time-01', busy: false } as any as GhRunner,
+      /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+      { id: '0002', name: 'keep-this-not-min-time-02', busy: false } as any as GhRunner,
```

Comment:
nit: The unsafe cast and lint suppression could be extracted into a simple function to avoid code duplication, something like this:
```
/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
const mockRunner(runnerDef:any): GhRunner = runnerDef as GhRunner;
```